asabiris/formatter/jinja/service.py

```diff
@@ -27,31 +26,54 @@ async def format(self, template_path, template_params):
 		template = jinja2.Template(template_io.read().decode('utf-8'))
 
 		# This creates nested dictionary from keys with dot notation
-		template_params = create_nested_dict_from_dots_in_keys(template_params)
+		template_params = self.create_nested_dict_from_dots_in_keys(template_params)
 		return template.render(**template_params)
 
-
-def create_nested_dict_from_dots_in_keys(data):
-	"""
-	This function creates a nested dictionary from a dictionary with keys containing dots.
-
-	:param data: The input dictionary that may contain keys with dots in them, indicating nested levels
-	of dictionaries
-	:return: a nested dictionary where keys containing dots (".") have been split into sub-dictionaries.
-	"""
-	nested_dict = {}
-	for key, value in data.items():
-		if isinstance(value, dict):
-			value = create_nested_dict_from_dots_in_keys(value)
-
-		if '.' in key:
-			parts = key.split('.')
-			current_dict = nested_dict
-			for part in parts[:-1]:
-				current_dict.setdefault(part, {})
-				current_dict = current_dict[part]
-
-			current_dict[parts[-1]] = value
-		else:
-			nested_dict[key] = value
-	return nested_dict
+	def create_nested_dict_from_dots_in_keys(self, data):
```

Comment:
it does not work for this dictionary:
```
	{
		"alert": {
			"event.id": 2
		},
		"happy.unicorn": {
			"best.friend": "Rabbit"
		}
	}
```
nor for this one:
```
	{
		"happy.unicorn": {
			"best.friends": {
				"rabbit": "Joe",
				"cat": "Ema",
				"praying.mantis": "Sisi"
			}
		}
	}
```
keys followed by dictionary as a value are not "parsed"